pkg/registries/ecr/transport.go

```diff
@@ -0,0 +1,64 @@
+package ecr
+
+import (
+	"encoding/base64"
+	"net/http"
+	"strings"
+	"time"
+
+	awsECR "github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/heroku/docker-registry-client/registry"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/registries/docker"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+type awsTransport struct {
+	registry.Transport
+	config    *docker.Config
+	client    *awsECR.ECR
+	expiresAt *time.Time
+	mutex     sync.Mutex
+}
+
+func newAWSTransport(config *docker.Config, client *awsECR.ECR) *awsTransport {
+	return &awsTransport{config: config, client: client}
+}
+
+func (t *awsTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+	if t.expiresAt == nil || t.expiresAt.After(time.Now()) {
+		err := t.refreshNoLock()
+		if err != nil {
+			return nil, err
+		}
+	}
+	resp, err := t.Transport.RoundTrip(req)
+	return resp, err
+}
+
+func (t *awsTransport) refreshNoLock() error {
+	authToken, err := t.client.GetAuthorizationToken(&awsECR.GetAuthorizationTokenInput{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get authorization token")
+	}
+	if len(authToken.AuthorizationData) == 0 {
+		return errors.Errorf("received empty authorization data in token: %s", authToken)
```

Comment:
Hmm, what other information is going to be there for the auth token besides the authorization data? Not sure we want to print this / add this to an error message?

Reply:
It doesn't contain any other data. We can just remove it.